### PR TITLE
Refactor project configuration

### PR DESCRIPTION
**Pull request by eemeli**
Move everything from `setup.cfg` and `setup.py` to `pyproject.toml`, and clean up `tox.ini`.

### CODE REVIEW

**Comment by eemeli** [comment]
To validate how this config is handled by build tools and PyPi, I've uploaded the current head of this PR here: https://test.pypi.org/project/compare-locales/9.0.1.1/

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.7",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
```
Comment (flodolo) [inline pyproject.toml:24]: Add 3.10, given we're testing it now and it's in tox.ini?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  "Topic :: Software Development :: Libraries :: Python Modules",
+  "Topic :: Software Development :: Localization",
+  "Topic :: Software Development :: Testing",
+]
+requires-python = ">=3.7, <4"
+dependencies = [
+  "fluent.syntax >=0.18.0, <0.20",
+  "six",                           # undeclared dependency of fluent-syntax 0.18.1
```
Comment (flodolo) [inline pyproject.toml:32]: Out of curiosity, did we fix this in 0.19?

**Review by flodolo — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (eemeli) [reply pyproject.toml:24]: D'oh, of course.

```suggestion
  "Programming Language :: Python :: 3.9",
  "Programming Language :: Python :: 3.10",
```

File: pyproject.toml (hunk 2 above)
Comment (eemeli) [reply pyproject.toml:32]: Yes.

**Pull request merged**